scripts/test_screeps_rl_experiment_card.py

```
@@ -242,6 +242,86 @@ def test_cli_generates_loop_a_supply_from_latest_accepted_dataset(self) -> None:
+        self.assertEqual(supply["state"], "available")
+        self.assertTrue(supply["available_for_training"])
+        self.assertIsNone(supply["consumed_at"])
+        self.assertIsNone(supply["consumed_by_report_id"])
+        self.assertTrue(card_helper.is_loop_a_card_available_for_training(generated))
+        for field in ("liveEffect", "officialMmoWrites", "officialMmoWritesAllowed"):
+            self.assertFalse(generated[field])
+            self.assertFalse(generated["safety"][field])
+
+            live_regression = json.loads(json.dumps(generated))
+            live_regression[field] = True
+            with self.assertRaises(card_helper.CardValidationError):
```

Ruff (0.15.12)

[warning] Use `pytest.raises` instead of unittest-style `assertRaises`

Replace `assertRaises` with `pytest.raises`

(PT027)